test/pg.ddl.unit.js

```diff
@@ -0,0 +1,788 @@
+          {
+            from: 'LocalCategory2',
+            to: 'LocalCategory1',
+            name: 'field',
+            link: ms.categories.get('LocalCategory2').definition.field,
+            destination: 'LocalCategory1',
+          },
+        ],
+        ms.categories
+      ),
+      `\
+ALTER TABLE "LocalCategory2" ADD CONSTRAINT "fkLocalCategory2field" \
+FOREIGN KEY ("field") REFERENCES "LocalCategory1" ("Id") \
+ON UPDATE RESTRICT ON DELETE RESTRICT;
+
+CREATE TABLE "CategoryWithManyfield" (
+  "CategoryWithMany" bigint NOT NULL,
+  "field"            bigint NOT NULL
+);
+ALTER TABLE "CategoryWithManyfield" \
+ADD CONSTRAINT "fkCategoryWithManyfieldCategoryWithMany" \
+FOREIGN KEY ("CategoryWithMany") REFERENCES "CategoryWithMany" ("Id") \
+ON UPDATE RESTRICT ON DELETE CASCADE;
+ALTER TABLE "CategoryWithManyfield" \
+ADD CONSTRAINT "fkCategoryWithManyfieldfield" \
+FOREIGN KEY ("field") REFERENCES "LocalCategory1" ("Id") \
+ON UPDATE RESTRICT ON DELETE CASCADE;
+ALTER TABLE "CategoryWithManyfield" \
+ADD CONSTRAINT "pkfieldCategoryWithManyLocalCategory1" \
+PRIMARY KEY ("CategoryWithMany", "field");`
+    )
+  );
+
+  unitTests.testSync('pg.ddl generateIndexes', test =>
+    test.strictSame(
+      ddl.generateIndexes(
+        [
+          { name: 'index1', property: { domain: 'Nomen', index: true } },
+          { name: 'index3', property: createIndex('index2', 'index3') },
+        ],
+        'Schema'
+      ),
+      `\
+CREATE INDEX "idxSchemaindex1" on "Schema" ("index1");
+CREATE INDEX "idxSchemaindex3" on "Schema" ("index2", "index3");`
+    )
+  );
+
+  unitTests.testSync('pg.ddl generateUnique', test =>
+    test.strictSame(
+      ddl.generateUnique(
+        [
+          { name: 'unique1', property: { domain: 'Nomen', unique: true } },
+          {
+            name: 'unique3',
+            property: createUnique('unique2', 'unique3'),
+          },
+        ],
+        'Schema'
+      ),
+      `\
+ALTER TABLE "Schema" ADD CONSTRAINT "akSchemaunique1" UNIQUE ("unique1");
+ALTER TABLE "Schema" ADD CONSTRAINT "akSchemaunique3" \
+UNIQUE ("unique2", "unique3");`
+    )
+  );
+
+  unitTests.testSync('pg.ddl generateId', test => {
+    test.strictSame(ddl.generateId('Local', 2), '"Id" bigserial');
+    test.strictSame(ddl.generateId('Global', 2), '"Id" bigint');
+  });
+
+  unitTests.testSync('pg.ddl getResolvalbeLinks', test => {
+    test.strictSame(
+      ddl.getResolvableLinks(
+        'Schema1',
+        new Map(),
+        [
+          {
+            name: 'field',
+            property: { category: 'Schema2' },
+            destination: 'Schema2',
+          },
+          {
+            name: 'field',
+            property: { category: 'Schema1' },
+            destination: 'Schema1',
+          },
+          {
+            name: 'field',
+            property: { category: 'Schema4' },
+            destination: 'Schema4',
+          },
+        ],
+        ['Schema1', 'Schema2']
+      ),
+      [
+        [
```

Comment:
I'd be better to extract this arrays to variables named `unresolved` and `resolvableLinks` accordingly.